pkg/river/internal/stdlib/stdlib.go

```diff
@@ -73,4 +73,18 @@ var Identifiers = map[string]interface{}{
 		}
 		return res, nil
 	},
+
+	"coalesce": value.RawFunction(func(funcValue value.Value, args ...value.Value) (value.Value, error) {
+		for _, arg := range args {
+			if !arg.Reflect().IsZero() {
+				if argType := value.RiverType(arg.Reflect().Type()); (argType == value.TypeArray || argType == value.TypeObject) && arg.Len() == 0 {
```

Comment:
I think we need to add some conditions for `null`

To handle `coalesce([],null)`, I doubt anyone will slam null in there but its possible exports from other components could be null. 

```
if arg.Type() == value.TypeNull {
				continue
			}
```